Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.CosmosRepository.Extensions;
+
+namespace Microsoft.Azure.CosmosRepository
+{
+    /// <inheritdoc/>
+    public class InMemoryRepository<TItem> : IRepository<TItem> where TItem : class, IItem
+    {
+        internal ConcurrentDictionary<string,TItem> Items { get; } = new();
+
+        /// <inheritdoc/>
+        public ValueTask<TItem> GetAsync(string id, string partitionKeyValue = null, CancellationToken cancellationToken = default)
+            => GetAsync(id, new PartitionKey(partitionKeyValue ?? id), cancellationToken);
+
+        /// <inheritdoc/>
+        public async ValueTask<TItem> GetAsync(string id, PartitionKey partitionKey, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            if (partitionKey == default)
+            {
+                partitionKey = new PartitionKey(id);
+            }
+
+            TItem item = Items.Values.FirstOrDefault(i => i.Id == id && new PartitionKey(i.PartitionKey) == partitionKey);
+
+            if (item is null)
+            {
+                NotFound();
+            }
+
+            return item is { Type: { Length: 0 } } || item.Type == typeof(TItem).Name ? item : default;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<IEnumerable<TItem>> GetAsync(Expression<Func<TItem, bool>> predicate, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            return Items.Values.Where(predicate.Compose(
+                item =>  item.Type == typeof(TItem).Name, Expression.AndAlso).Compile());
+        }
+
+        /// <inheritdoc/>
+        public ValueTask<IEnumerable<TItem>> GetByQueryAsync(string query, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
```

Comment:
I know this was already discussed on the call, but worth adding a backlog ticket to implement this? Not sure if this would be helpful: https://github.com/axelheer/nein-linq#lambda-injection

Reply:
yeah I am more than happy to add this as a work item.